x-pack/plugin/inference/qa/test-service-plugin/src/main/java/org/elasticsearch/xpack/inference/mock/TestDenseInferenceServiceExtension.java

```diff
@@ -138,41 +140,78 @@ public void chunkedInfer(
 
         private InferenceTextEmbeddingFloatResults makeResults(List<String> input, int dimensions) {
             List<InferenceTextEmbeddingFloatResults.InferenceFloatEmbedding> embeddings = new ArrayList<>();
-            for (int i = 0; i < input.size(); i++) {
-                float[] doubleEmbeddings = generateEmbedding(input.get(i), dimensions);
-                List<Float> floatEmbeddings = new ArrayList<>(dimensions);
-                for (int j = 0; j < dimensions; j++) {
-                    floatEmbeddings.add(doubleEmbeddings[j]);
-                }
+            for (String inputString : input) {
+                List<Float> floatEmbeddings = generateEmbedding(inputString, dimensions);
                 embeddings.add(InferenceTextEmbeddingFloatResults.InferenceFloatEmbedding.of(floatEmbeddings));
             }
             return new InferenceTextEmbeddingFloatResults(embeddings);
         }
 
         private List<ChunkedInferenceServiceResults> makeChunkedResults(List<String> input, int dimensions) {
-            var chunks = new ArrayList<InferenceTextEmbeddingFloatResults.InferenceFloatEmbedding>();
-            for (int i = 0; i < input.size(); i++) {
-                float[] embedding = generateEmbedding(input.get(i), dimensions);
-                chunks.add(new InferenceTextEmbeddingFloatResults.InferenceFloatEmbedding(embedding));
-            }
-
-            return InferenceChunkedTextEmbeddingFloatResults.listOf(input, new InferenceTextEmbeddingFloatResults(chunks));
+            InferenceTextEmbeddingFloatResults nonChunkedResults = makeResults(input, dimensions);
+            return InferenceChunkedTextEmbeddingFloatResults.listOf(input, nonChunkedResults);
```

Comment:
Simplified the logic here. `makeChunkedResults` does essentially the same thing as `makeResults`, except that the former assigns an embedding to each input.